File: data/EXP01/golang_no_system_var/2442/2442.output_3.go
func countDistinctIntegers(nums []int) int {
    distinctIntegers := make(map[int]bool)

    for _, num := range nums {
        distinctIntegers[num] = true
        numStr := strconv.Itoa(num)
        reversed := reverse(numStr)
        reversedNum, _ := strconv.Atoi(reversed)
        distinctIntegers[reversedNum] = true
    }

    return len(distinctIntegers)
}

func reverse(str string) string {
    reversed := ""
    for i := len(str) - 1; i >= 0; i-- {
        reversed += string(str[i])
    }
    return reversed
}

File: data/EXP01/python/2442/2442.output_3.py
# @lc app=leetcode id=2442 lang=python3
class Solution:
    def countDistinctIntegers(self, nums: List[int]) -> int:
        distinct_integers = set()

        for num in nums:
            reverse_num = int(str(num)[::-1])
            distinct_integers.add(reverse_num)

        return len(distinct_integers)